uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-widget/leave-widget-absence-types-amount-taken.component.js

```diff
@@ -0,0 +1,122 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'leave-absences/shared/components/leave-requests-heatmap.component',
+  'leave-absences/shared/models/leave-request.model'
+], function (_, moment, components) {
+  components.component('leaveWidgetAbsenceTypesAmountTaken', {
+    bindings: {
+      title: '@',
+      absenceTypes: '<',
+      contactId: '<',
+      absencePeriod: '<',
+      leaveRequestStatuses: '<'
+    },
+    controller: absenceTypesTakenController,
+    controllerAs: 'absenceTypesTaken',
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-widget/leave-widget-absence-types-amount-taken.html';
+    }]
+  });
+
+  absenceTypesTakenController.$include = ['$scope', 'LeaveRequest',
+    'OptionGroup'];
+
+  function absenceTypesTakenController ($scope, LeaveRequest, OptionGroup) {
+    var vm = this;
+
+    vm.leaveRequests = [];
+
+    vm.$onChanges = $onChanges;
+
+    /**
+     * Initializes the controller by emiting a child is loading event.
+     */
+    (function init () {
+      $scope.$emit('LeaveWidget::childIsLoading');
+    })();
+
+    /**
+     * Implements the $onChanges method for angular controllers. When bindings
+     * are ready for use, it loads all dependencies.
+     */
+    function $onChanges () {
```

Comment:
That approach is good when I want something to happen when ONE binding is updated, not so much for multiple bindings since they might not be passed at the same time (promises, etc.)